File: backend/food-tracker-api/src/main/java/com/foodtracker/foodtrackerapi/common/utils/DateUtils.java
package com.foodtracker.foodtrackerapi.common.utils;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

/**
 * Utility class to convert the incoming date string into the correct date
 * format before it is processed.
 * 
 * @author Janina Mattes
 */

public class DateUtils {

    public static Date addDays(Date date, int days)
    {
        Calendar cal = Calendar.getInstance();
        cal.setTime(date);
        cal.add(Calendar.DATE, days); //minus number would decrement the days
        return cal.getTime();
    }

    public static String convertDateFormat(String dateStr){
        // formats a date string into the correct format
        DateFormat inputFormat = new SimpleDateFormat("yyyy-MM-dd");
        DateFormat outputFormat = new SimpleDateFormat("dd.MM.yyyy");
      
        Date inputDate  = new Date();
        try {
            inputDate = inputFormat.parse(dateStr);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        String outputDateStr = outputFormat.format(inputDate);
        return outputDateStr;
    }
}

File: frontend/android-feedme-app/app/src/main/java/com/mobilesystems/feedme/ui/common/utils/DateUtils.kt
package com.mobilesystems.feedme.ui.common.utils

import android.icu.text.DateFormat
import android.icu.text.SimpleDateFormat
import android.net.ParseException
import java.util.*
import java.util.concurrent.TimeUnit

fun getTimeDiff(dateStr: String): Long {
    var result: Long = 0
    if(dateStr.isNotEmpty()) {
        val myFormat = "dd.MM.yyyy"
        val sdf = SimpleDateFormat(myFormat, Locale.GERMANY)
        val endDateValue: Date = sdf.parse(dateStr)
        val startDateValue = Calendar.getInstance().timeInMillis
        result = TimeUnit.MILLISECONDS.toDays(endDateValue.time - startDateValue)
    }
    return result
}

fun addDaysToCurrentDate(days: Int): Calendar {
    val currentDate = getCurrentDate()
    currentDate.add(Calendar.DATE, days)
    return currentDate
}

fun getCurrentDate(): Calendar {
    val calendar =  Calendar.getInstance()
    val currentDate = calendar.time
    calendar.time = currentDate
    return calendar
}

fun convertDateFormat(dateStr: String): String {
    // formats a date string into the correct format
    val inputFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd")
    val outputFormat: DateFormat = SimpleDateFormat("dd.MM.yyyy")
    var inputDate = Date()
    try {
        inputDate = inputFormat.parse(dateStr)
    } catch (e: ParseException) {
        e.printStackTrace()
    }
    return outputFormat.format(inputDate) ?: dateStr
}

fun reconvertDateFormat(dateStr: String): String{
    val inputFormat: DateFormat =SimpleDateFormat("dd.MM.yyyy")
    val outputFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd")
    var inputDate = Date()
    try {
        inputDate = inputFormat.parse(dateStr)
    } catch (e: ParseException) {
        e.printStackTrace()
    }
    return outputFormat.format(inputDate) ?: dateStr
}

fun convertStringToDate(dateStr: String): Date {
    val sdf = SimpleDateFormat("dd.MM.yyyy")
    val date = sdf.parse(dateStr)
    return date
}
